### PR TITLE
Upgrade to latest for all linting deps

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

I have also condenced the eslint rules to join together the rules for `.tsx` and `.ts` files since they should always have the same rules and `jsx: true` does nothing according to the eslint documentation for `.tsx` and `.jsx` files anyway.

I have also changed all the `.d.ts` files into `.ts` files as vscode doesn't do auto complete within `.d.ts` files.

This PR removes some mobx decorators, namely when they were used on function fields as the indent eslint rule is broken for function fields with annotations. They were replaced by the higher order function calls instead.

### CODE REVIEW

File: src/renderer/components/icon/icon.tsx [hunk 1]
```diff
@@ -59,10 +60,7 @@ export class Icon extends React.PureComponent<IconProps> {
 
         // fallthrough
       case "Enter": {
-        // eslint-disable-next-line react/no-find-dom-node
-        const icon = findDOMNode(this) as HTMLElement;
-
-        setTimeout(() => icon.click());
+        setImmediate(() => this.ref?.click());
```
Comment (aleksfront) [inline src/renderer/components/icon/icon.tsx]: We should omit `setImmediate`. It works without it and `setImmediate` doesn't have enough browser support https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate

File: src/renderer/components/icon/icon.tsx [hunk 2]
```diff
@@ -31,6 +30,8 @@ export interface IconProps extends React.HTMLAttributes<any>, TooltipDecoratorPr
 
 @withTooltip
 export class Icon extends React.PureComponent<IconProps> {
+  public ref?: HTMLElement;
```
Comment (aleksfront) [inline src/renderer/components/icon/icon.tsx]: We can use `React.creareRef` here https://reactjs.org/docs/refs-and-the-dom.html

Hovewer, I don't see any behavior difference.

**Comment by aleksfront** [comment]
Also, "removing mobx decorations" should be mentioned in description.

File: src/renderer/components/icon/icon.tsx (hunk 2 above)
Comment (Nokel81) [reply src/renderer/components/icon/icon.tsx]: I'll use createRef

**Comment by Nokel81** [comment]
@aleksfront PTALA

**Review by aleksfront — APPROVED** [review]
LGTM

File: src/extensions/extensions-store/extensions-store.ts [hunk 3]
```diff
@@ -42,10 +42,9 @@ export class ExtensionsStore extends BaseStore<LensExtensionsStoreModel> {
...
   }
 
-  @action
-  mergeState = (extensionsState: Record<LensExtensionId, LensExtensionState>) => {
+  mergeState = action((extensionsState: Record<LensExtensionId, LensExtensionState>) => {
     this.state.merge(extensionsState);
-  };
+  });
 
   @action
```
Comment (jim-docker) [inline src/extensions/extensions-store/extensions-store.ts:49]: Is it weird now to to use decorator sometimes? Guess it depends on context

File: src/renderer/components/+workloads-pods/pod-details-secrets.tsx [hunk 4]
```diff
@@ -20,19 +20,22 @@ interface Props {
...
+            namespace: pod.getNs(),
+          })),
+        );
 
-    secrets.forEach(secret => secret && this.secrets.set(secret.getName(), secret));
-  });
+        secrets.forEach(secret => secret && this.secrets.set(secret.getName(), secret));
+      }),
+    ]);
+  }
```
Comment (jim-docker) [inline src/renderer/components/+workloads-pods/pod-details-secrets.tsx:38]: Isn't it acceptable to just add the indentation that linting demands, i.e.
```
  @disposeOnUnmount
    secretsLoader = autorun(async () => {
      const { pod } = this.props;

      const secrets = await Promise.all(
        pod.getSecrets().map(secretName => secretsApi.get({
          name: secretName,
          namespace: pod.getNs(),
        })),
      );

      secrets.forEach(secret => secret && this.secrets.set(secret.getName(), secret));
    });
```

It feels odd to be able to use decorators in some instances but not in others. 

Is this just a bug with eslint?

File: yarn.lock [hunk 5]
```diff
@@ -12,13 +12,6 @@
   resolved "https://registry.yarnpkg.com/@async-fn/jest/-/jest-1.5.3.tgz#42be6c0e8ba5ccd737e006ca600e7e319fe2a591"
   integrity sha512-iQ9gAPZFW5U6TNcFS99ffwYYsB9LNecTnvG73BaDc/zAD0qOWctY1imEACC1pLymmm/xaf/OUq9I9QenfkatTA==
```
Comment (jim-docker) [inline yarn.lock:14]: is this large diff expected?

**Review by jim-docker — APPROVED** [review]

File: src/extensions/extensions-store/extensions-store.ts (hunk 3 above)
Comment (Nokel81) [reply src/extensions/extensions-store/extensions-store.ts:49]: The decorators are useful for methods as opposed to field functions

File: yarn.lock (hunk 5 above)
Comment (Nokel81) [reply yarn.lock:14]: Yes because I updated dependencies

**Pull request merged**